src/map/entities/charentity.cpp

```diff
@@ -602,6 +602,23 @@ int8 CCharEntity::getShieldSize()
     return PItem->getShieldSize();
 }
 
+int16 CCharEntity::getShieldDefense()
+{
+    CItemEquipment* PItem = getEquip(SLOT_SUB);
+
+    if (PItem == nullptr)
+    {
+        return 0;
+    }
+
+    if (!PItem->IsShield())
+    {
+        return 0;
+    }
+
+    return PItem->getModifier(Mod::DEF);
+}
```

Comment:
int16 CCharEntity::getShieldDefense()
{
    CItemEquipment* PItem = getEquip(SLOT_SUB);

    if (PItem && PItem->IsShield())
    {
        return PItem->getModifier(Mod::DEF);
    }

    return 0;
}

Reply:
Refactored.

Reply:
I think you swaped both functions. This is the charentity file and that is the lua base entity function

Reply:
I definitely did, fixed.